core/src/main/java/org/apache/james/core/User.java

```diff
@@ -76,10 +79,17 @@ public static User from(String localPart, Optional<String> domain) {
     private final String localPart;
     private final Optional<Domain> domainPart;
 
+    private boolean isLocalPartValid(String localPart) {
+        List<String> localParts = ImmutableList.copyOf(Splitter.on(CharMatcher.anyOf(ILLEGAL_USERNAME_CHARACTERS))
```

Comment:
Why a `copyOf`?

You also do not need a temporary variable.

Maybe you can simply try to search twice for these characters instead of doing the full split.